types/comment.ts

```diff
@@ -50,6 +60,40 @@ export class Comment extends LazyObject {
     });
   }
 
+  /** likedBy
+   * Checks if the given user has liked this comment
+   */
+  public async likedBy(user: User) {
+    return this.getLikes().then((likes) =>
+      likes.some((like) => refEqual(like.docRef!, user.docRef!))
+    );
+  }
+
+  /** addLike
+   * Add a like to this comment
+   */
+  public async addLike(user: User) {
+    if (this.hasData && (await this.likedBy(user))) return;
+    await runTransaction(db, async (transaction) => {
+      transaction.update(this.docRef!, { likes: arrayUnion(user.docRef!) });
+    });
+    if (this.hasData) this.likes?.push(user);
+  }
+
+    /** removeLike
+   * Remove a like from this comment
+   */
+  public async removeLike(user: User) {
+    if (this.hasData && !(await this.likedBy(user))) return;
+    await runTransaction(db, async (transaction) => {
+      transaction.update(this.docRef!, { likes: arrayRemove(user.docRef!) });
+    });
```

Comment:
Same here

Reply:
thanks for the help, just pushed & it's ready for another review